### PR TITLE
feat: configure client certificate and public network access

### PR DESCRIPTION
**Pull request by Adnan029**
We used this module internally and made some changes which need to be compliant as per internal compliance policy.

Below are the changes -

**log_analytics_workspace_id** was mandate before now we have added a variable to handle it if this can be used or can be skip using the following  **logs_analytics_enable**.

We have added 2 more attributes to the webapp **client_certificate_mode & client_certificate_enabled** and we parameterized this attribute the **public_network_access_enabled**.

We have done a testing with these changes and it is working as expected.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -316,3 +316,22 @@ variable "storage_accounts" {
     error_message = "Storage account type must be either \"AzureFiles\" or \"AzureBlob\"."
   }
 }
+
+variable "public_network_access_enabled" {
+    description = "Should public access be enabled for this Web App?"
+    default = true
+    type = bool
+}
```
Comment (hknutsen) [inline variables.tf:324]: - Update order of arguments
- Update description
- Update formatting

```suggestion
variable "public_network_access_enabled" {
    description = "Should public network access be enabled for this Web App?"
    type        = bool
    default     = true
}
```

File: variables.tf [hunk 2]
```diff
@@ -316,3 +316,22 @@ variable "storage_accounts" {
...
+
+
+variable "client_certificate_mode" {
+  description = "Should client cerftificate mode be enabled for this Web App?"
+  default = "Optional"
+  type = string
+}
```
Comment (hknutsen) [inline variables.tf:331]: - Update order of arguments
- Update description
- Update default value (feel free to argue against this change)
- Update formatting

```suggestion
variable "client_certificate_mode" {
  description = "The client cerftificate mode for this Web App. Value must be \"Required\", \"Optional\" or \"OptionalInteractiveUser\"."
  type        = string
  default     = "Required"
}
```

File: variables.tf [hunk 3]
```diff
@@ -316,3 +316,22 @@ variable "storage_accounts" {
...
+
+variable "client_certificate_enabled" {
+  description = "Should client certificate be enabled for this Web App?"
+  default = false
+  type = bool
+}
```
Comment (hknutsen) [inline variables.tf:337]: - Update order of arguments
- Update formatting

```suggestion
variable "client_certificate_enabled" {
  description = "Should client certificate be enabled for this Web App?"
  type        = bool
  default     = false
}
```

**Review by hknutsen — CHANGES_REQUESTED** [review]
Changes related to client certificate and public network access look good 😄 🙌 Added a few suggestions for minor improvements. We'll also have to implement the same changes for the `azurerm_windows_web_app.this` resource.

In regards to the Log Analytics change: Is the goal to allow users to prevent the creation of the `azurerm_monitor_diagnostic_setting.this` resource? If yes, what's the reason for this request? (please note that this is a public repository, so no company specific information should be shared here. Feel free to contact me directly to explain your case.)

**Comment by Adnan029** [comment]
Thanks for the swift response @hknutsen. 

I have committed your suggestion and its looks good to me. Also, client certificate and public network access changes already applied to `azurerm_windows_web_app.this`.

In regards to the Log Analytics change: Is the goal to allow users to prevent the creation of the azurerm_monitor_diagnostic_setting.this resource? If yes, what's the reason for this request? (please note that this is a public repository, so no information related specifically to your company should be shared here. Feel free to contact me directly to explain your case.)

Yes, the goal is to allow users to prevent the creation of the azurerm_monitor_diagnostic_setting.this resource because we aim to provide flexibility in scenarios where logging is not required or desired.

**Comment by hknutsen** [comment]
> [...] client certificate and public network access changes already applied to azurerm_windows_web_app.this.

Didn't notice in my initial review, but yes, this looks to already be in place.

> Yes, the goal is to allow users to prevent the creation of the azurerm_monitor_diagnostic_setting.this resource because we aim to provide flexibility in scenarios where logging is not required or desired.

I think it would be a good idea to remove the changes to related to the diagnostic setting in this PR, then create a separate PR where we can continue to discuss a potential implementation.

**Comment by hknutsen** [comment]
I'll create a quick PR with a suggested implementation for conditional diagnostic setting.

**Comment by Adnan029** [comment]
Really appreciate your prompt response @hknutsen. I have removed log diagnostic setting.

**Review by hknutsen — APPROVED** [review]

**Pull request merged**